elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java

```diff
@@ -56,15 +58,30 @@ public ElasticsearchSchemaFactory() {
 
     try {
       final String coordinatesString = (String) map.get("coordinates");
-      Preconditions.checkState(coordinatesString != null,
-          "'coordinates' is missing in configuration");
 
-      final Map<String, Integer> coordinates = mapper.readValue(coordinatesString,
-          new TypeReference<Map<String, Integer>>() { });
+      final String hostsString = (String) map.get("hosts");
 
-      // create client
-      final RestClient client = connect(coordinates);
+      final Set<HttpHost> hostSet = new LinkedHashSet<>();
+
+      if (coordinatesString != null) {
+        final Map<String, Integer> coordinates = mapper.readValue(coordinatesString,
+                new TypeReference<Map<String, Integer>>() { });
+
+        coordinates.forEach((hostName, port) ->  hostSet.add(new HttpHost(hostName, port)));
```

Comment:
instead of modifying the state (`hosts`) variable (using `forEach`) can you use streams ? 
```java
hosts = coordinates.entrySet()
.stream(entry -> new HttpHost(entry.getKey(), entry.getValue())).collect(Collectors.toList());
```

Reply:
Done